lego/apps/events/tasks.py

```diff
@@ -477,7 +479,8 @@ def bump_waiting_users_to_new_pool(self, logger_context=None):
         with transaction.atomic():
             locked_event = Event.objects.select_for_update().get(pk=event_id)
             if locked_event.waiting_registrations.exists():
-                for pool in locked_event.pools.all():
+                locked_pools = locked_event.pools.select_for_update().all()
+                for pool in locked_pools:
```

Comment:
The locking order here should be fine, since we check for `pool.is_full` below. Had we had `event.is_full` (when merged) this would cause race conditions as well.

We should be wary to make sure we have this order as well everywhere, otherwise we could end up with deadlocks :grimacing: